### ISSUE TITLE
How to pad input sequences for implementing CuDNN LSTM in TF2.0?

### ISSUE THREAD

**Issue by rishabhsahrawat** [tensorflow/tensorflow#30745]
For implementing CuDNN LSTM, according to [docs](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/layers/LSTM) there are 6 requirements as follows- 

> activation == 'tanh'
recurrent_activation == 'sigmoid'
recurrent_dropout == 0
unroll is False
use_bias is True
Inputs are not masked or strictly right padded.

According to my understanding, the last one say to have input sequences not right padded. However, in text classification I want to pad sequences and I am using [the function](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/data/Dataset#padded_batch), now I do not know how to left pad using this function, as I could check it always right pad the input sequences, which means one can never use CuDNN LSTM for training on GPU(s) if they are padding the inputs. However, it can be done easily using `tf.keras..pad_sequences` function, defined [here](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/preprocessing/sequence/pad_sequences).

**Comment by jkamalu** [comment]
I think you are misreading the requirement. Inputs should be either [not masked] or [strictly right-padded]. For reference, see the following function used to determine CuDNN compatibility: https://github.com/tensorflow/tensorflow/blob/8e423e3d56390671f0d954c90f4fd163ab02a9c1/tensorflow/python/keras/layers/recurrent_v2.py#L1182

Note: the LSTM with CuDNN is experiencing a lot of bug reports atm (e.g. in loops, in distributed strategies, etc.), all of which are open issues as of right now. However, concerning your question, I think you can close this issue.

**Comment by rishabhsahrawat** [comment]
@jkamalu thank you for your answer and clarification. I still have a question about how to left pad in TF2.0, without using `tf.keras..pad_sequences` function. Is it possible?

-Rishabh Sahrawat

**Comment by jkamalu** [comment]
Use tf.pad. Choose the constant mode and parameterize the padding on the left-hand side with the max desired length and current length of the sequence and use zero padding on the right side. Play around with eager tensors to get this right if it doesn't seem obvious from the documentation. 

https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/pad

 If you want your input data padded, use tf.pad with the Dataset map function during data feeding. 

https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/data/Dataset#map

However, in my experience, this adds unnecessary overhead and can bottleneck training... So, what you should do if you can is preprocess all sequences to be padded however you like, save them to disk, and then load them as such during Dataset creation/iteration.

**Comment by rishabhsahrawat** [comment]
@jkamalu thank you for your answer. I appreciate your help.
